pkg/rekor/verifier.go

```diff
@@ -40,3 +49,118 @@ func GetLogVerifier(ctx context.Context, rekorClient *client.Rekor) (signature.V
 	}
 	return verifier, nil
 }
+
+// verifyLatestCheckpoint fetches and verifies the signature of the latest checkpoint from log info fetched from Rekor.
+// If it successfully verifies the checkpoint's signature, it returns the checkpoint; otherwise, it returns an error.
+func verifyLatestCheckpointSignature(logInfo *models.LogInfo, verifier signature.Verifier) (*util.SignedCheckpoint, error) {
+	checkpoint := &util.SignedCheckpoint{}
+	if err := checkpoint.UnmarshalText([]byte(*logInfo.SignedTreeHead)); err != nil {
+		return nil, fmt.Errorf("unmarshalling logInfo.SignedTreeHead to Checkpoint: %v", err)
+	}
+	if !checkpoint.Verify(verifier) {
+		return nil, fmt.Errorf("verifying checkpoint (size %d, hash %s) failed", checkpoint.Size, hex.EncodeToString(checkpoint.Hash))
+	}
+	return checkpoint, nil
+}
+
+// verifyCheckpointConsistency reads and verifies the consistency of the previous latest checkpoint from a log info file against the current up-to-date checkpoint.
+// If it successfully fetches and verifies the consistency between these two checkpoints, it returns the previous checkpoint; otherwise, it returns an error.
+func verifyCheckpointConsistency(logInfoFile *string, checkpoint *util.SignedCheckpoint, treeID string, rekorClient *client.Rekor, verifier signature.Verifier) (*util.SignedCheckpoint, error) {
+	var prevCheckpoint *util.SignedCheckpoint
+	prevCheckpoint, err := file.ReadLatestCheckpoint(*logInfoFile)
+	if err != nil {
+		return nil, fmt.Errorf("reading checkpoint log: %v", err)
+	}
+	if !prevCheckpoint.Verify(verifier) {
+		return nil, fmt.Errorf("verifying checkpoint (size %d, hash %s) failed", checkpoint.Size, hex.EncodeToString(checkpoint.Hash))
+	}
+	if err := verify.ProveConsistency(context.Background(), rekorClient, prevCheckpoint, checkpoint, treeID); err != nil {
+		return nil, fmt.Errorf("failed to verify log consistency: %v", err)
+	}
+	fmt.Fprintf(os.Stderr, "Root hash consistency verified - Current Size: %d Root Hash: %s - Previous Size: %d Root Hash %s\n",
+		checkpoint.Size, hex.EncodeToString(checkpoint.Hash), prevCheckpoint.Size, hex.EncodeToString(prevCheckpoint.Hash))
+	return prevCheckpoint, nil
+}
+
+// RunConsistencyCheck periodically verifies the root hash consistency of a Rekor log.
+func RunConsistencyCheck(interval *time.Duration, rekorClient *client.Rekor, verifier signature.Verifier, logInfoFile *string, mvs identity.MonitoredValues, outputIdentitiesFile *string, once *bool) error {
```

Comment:
We can do this as a follow up to keep the PR changes smaller (can track with a TODO), but I'd have this take in string and bool values rather than pointers. The code calling `RunConsistencyCheck` should check that the various flag values are set. This also makes it simpler to test as you can assume the value is always set rather than it maybe being null and crashing on dereference.